azure-pipelines-product-construction-service.yml

```diff
@@ -55,7 +57,7 @@ stages:
     - powershell: |
         Write-Host "Dev branch suffix is $(devBranchSuffix)"
         $shortSha = "$(Build.SourceVersion)".Substring(0,10)
-        $newDockerTag = "$(Build.BuildNumber)-$shortSha$(devBranchSuffix)"
+        $newDockerTag = "$(Build.BuildNumber)$(System.JobAttempt)-$shortSha$(devBranchSuffix)"
```

Comment:
I think in other repos we separate build number and job attempt via a dot.

Reply:
Oh, but still, is there some delimiter between the two in the resulting tag? Maybe if you gave an example of the tag it would be clearer for me at least

Reply:
2024061415-2fd735d544-dev, this is how the current tag looks,
20240614151-2fd735d544-dev this is how it would look with the new change

Reply:
Honestly, the most important part is the commit sha, and maybe the beginning of the date

Reply:
Then maybe its worth getting rid of the build number and attempt? That, or maybe it could be made part of the image's metadata as its good to have, but not in the tag

Reply:
if we did that we'd only be able to deploy once from a commit, since we'd already have an image with the same tag in the repository

Reply:
Sure, that makes sense.

Maybe then `2024061415-1-2fd735d544-dev` would be better? Otherwise its a bit confusing why the build number has an extra digit

Reply:
let's try, it might be over the char limit tho